sthype/hypergraph/hypergraph_from_spatial_graphs.py

```diff
@@ -64,17 +80,20 @@ def graph_segmentation(
         Edge have attribute center and there initial edge as a set : {node1, node2}.
         Node have attribute position
     """
-    label = max(spatial_graph.nodes) + 1
+    label: int = max(spatial_graph.nodes) + 1
     graph_segmented = nx.Graph()
 
-    nodes_position = {}
+    nodes_position: dict[int, Point] = {}
+
+    node1: int
+    node2: int
     for node1, node2 in spatial_graph.edges:
         pixels = spatial_graph.edge_pixels(node1, node2)
-        new_nodes = [
+        new_nodes: list[Point] = [
             pixels.line_interpolate_point(i * segments_length)
             for i in range(int(-(-pixels.length // segments_length) + 1))
         ]
-        new_edges_center = [
+        new_edges_center: list[Point] = [
             pixels.line_interpolate_point((i + 0.5) * segments_length)
             for i in range(int(-(-pixels.length // segments_length)))
         ]
```

Comment:
Use a better range : Calculate the needed number of segment. The put the new_nodes at the same distance interval : length / number of segment